src/components/Workspace.jsx

```diff
@@ -0,0 +1,258 @@
+import * as freddieEyes from '../assets/animations/freddieEyes.json';
+import * as freddieIdle from '../assets/animations/freddieIdle.json';
+import * as freddieSmall from '../assets/animations/freddieSmall.json';
+import * as freddieThumbs from '../assets/animations/freddieThumbs.json';
+import * as freddieSpin from '../assets/animations/freddieSpin.json';
+
+import * as lucyBows from '../assets/animations/lucyBows.json';
+import * as lucyBowSpin from '../assets/animations/lucyBowSpin.json';
+import * as lucyEyes from '../assets/animations/lucyEyes.json';
+import * as lucyIdle from '../assets/animations/lucyIdle.json';
+import * as lucyJump from '../assets/animations/lucyJump.json';
+import * as lucySmall from '../assets/animations/lucySmall.json';
+
+
+
+
+import Util from '../assets/util';
+
+class Workspace extends Component {
+    constructor() {
+        super();
+        this.state = {
+            shouldRenderSound: false,
+            songArray: [
+                freddieA,
+                geoffB,
+                lucyC,
+                freddieD,
+                lucyE,
+                geoffD,
+                geoffC,
+                lucyA,
+                freddieB,
+            ],
+            currentSong: null,
+            index: 0,
+            isStopped: false,
+            isPaused: false,
+            animationsArray: null,
+            currentLucyAnimation: lucyIdle,
+            currentFreddieAnimation: freddieIdle,
+            currentGeoffAnimation: geoffIdle,
+            animationsList: [
+                lucyBows,
+                lucyBowSpin,
+                lucyEyes,
+                lucyJump,
+                lucySmall,
+                lucyIdle,
+                geoffGlasses,
+                geoffJump,
+                geoffSpin,
+                geoffBig,
+                geoffCrown,
+                geoffIdle,
+                freddieDrip,
+                freddieEyes,
+                freddieSmall,
+                freddieSpin,
+                freddieThumbs,
+                freddieIdle,
+            ]
+        };
+    }
+
+    getArrayAndHandleSound() {
+        let animations = Util.createAnimationsArray(this.state.songArray);
+        this.setState({
+            animationsArray: animations,
+            index: 0,
+        }, () => {
+            this.handlePlaySound();
+        });
+    }
+
+    handlePlaySound() {
+        this.setState({
+            shouldRenderSound: true,
+            currentSong: this.state.songArray[this.state.index],
+            index: this.state.index + 1,
+        }, () => {
+            console.log(this.state)
+        });
+    }
+
+    handleSongFinishedPlaying = () => {
+        if (this.state.index <= this.state.songArray.length - 1) {
+            this.setState({
+                currentSong: this.state.songArray[this.state.index],
+                index: this.state.index + 1,
+                currentLucyAnimation: lucyIdle,
```

Comment:
do we need to set them to idle if its going to play the next animation? or just in case that doesn't work?

Reply:
they get set to idle so they dont dance during the other blob's sounds. if they don't reset to idle, they end up repeating the dance until they have another note